.github/buildomat/jobs/clippy.sh

```diff
@@ -29,3 +29,4 @@ ptime -m bash ./tools/install_builder_prerequisites.sh -y
 
 banner clippy
 ptime -m cargo xtask clippy
+ptime -m cargo doc
```

Comment:
Is this intentional?

Reply:
Yeah, I corrected several errors from `cargo doc` in this PR, so I figured I would add this to the clippy job so it catches such issues in CI.

Reply:
I thought we already had a thing for this?
https://github.com/oxidecomputer/omicron/blob/b5e62f612f6e629904d7b0db4d189685371a1e7a/.github/workflows/rust.yml#L74-L96